libraries/lib-wave-track/WaveClip.h

```diff
@@ -429,8 +446,20 @@ class WAVE_TRACK_API WaveClip final :
    void SetName(const wxString& name);
    const wxString& GetName() const;
 
+   // TimeToSamples and SamplesToTime take clip stretch ratio into account.
+   // Use them to convert time / sample offsets.
    sampleCount TimeToSamples(double time) const noexcept;
    double SamplesToTime(sampleCount s) const noexcept;
+   /*!
+    * Given the possibility of some clips of a track being time-stretched, and
+    * our approach to render to the track's sample rate only upon playback, the
+    * samples of clips may lie anywhere on the time track, not just at multiples
+    * of the track's sample period.
+    * Nevertheless, we maintain play start and end times, which define the
+    * visible boundaries of clips, on this sample rate grid. This is sufficient
+    * precision and preserves a little spacing between clips.
+    */
+   double SnapToTrackSample(double time) const noexcept;
```

Comment:
This function could be private, and instead `GetPlayStartTime` and `GetPlayEndTime` can be commented, that the return values are always multiple of the inverse of the track's stored rate, but independent of tempo or stretch ratio.

Reply:
All good suggestions.